lib/insights.js

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+'use strict';
+
+const debug = require('debug')('appinsights');
+
+function createWrappedClient(propertiesToInsert, client) {
+  const c = {};
+  if (client) {
+    c.trackEvent = addProperties.bind(client, client.trackEvent, propertiesToInsert, 1, true);
```

Comment:
Take a look at the library source code here (it is TS): https://github.com/Microsoft/ApplicationInsights-node.js/blob/master/Library/Client.ts#L49

For the `trackEvent` method, it is technically possible to have the following parameters:

- name
- properties
- measurements

Measurements are additional metrics. So in the case of that use of the function, which we have used in Witness Jobs and some other spots, the `properties` are not actually the last argument to the function, but instead the 2nd argument (index position 1).

This code is essentially [monkey patching](https://en.wikipedia.org/wiki/Monkey_patch) and injecting a little logic (a new property or properties) along the way.